### PR TITLE
Added ability to Enable/Disable Managed Platform Updates

### LINKED ISSUE #176: ElasticBeanstalk Recipe: Enable Managed Updates by default
**Is your feature request related to a problem? Please describe.**
The Elastic Beanstalk Recipe should enable Managed Updates by default.  This is the default behavior in the AWS Console.

image

Managed updates automatically install security updates without any user intervention.

**Describe the solution you'd like**
When I run the Elastic Beanstalk recipe, I'd like Managed Updates to be on for my normally deployed application.

If I need to disable it, I'd like an advanced option to disable it.

---

This is a :rocket: feature request

### PR DESCRIPTION
**Pull request by philasmar**
*Issue #, if available:*
Closes #176 

*Description of changes:*
PR199-After

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppElasticBeanstalkLinux/Configurations/ElasticBeanstalkManagedPlatformUpdatesConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+namespace AspNetAppElasticBeanstalkLinux.Configurations
+{
+    public class ElasticBeanstalkManagedPlatformUpdatesConfiguration
+    {
+        public bool ManagedActionsEnabled { get; set; } = true;
+        public string PreferredStartTime { get; set; } = "Sun:00:00";
```
Comment (ganeshnj) [inline src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppElasticBeanstalkLinux/Configurations/ElasticBeanstalkManagedPlatformUpdatesConfiguration.cs:9]: Can we use DateTime?

File: src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppElasticBeanstalkLinux/AppStack.cs [hunk 2]
```diff
@@ -88,9 +88,48 @@ internal AppStack(Construct scope, RecipeConfiguration<Configuration> recipeConf
...
+            if (settings.ElasticBeanstalkManagedPlatformUpdates.ManagedActionsEnabled)
+            {
+                var beanstalkrole = new Role(this, "BeanstalkServiceRole", new RoleProps
+                {
+                    AssumedBy = new ServicePrincipal("elasticbeanstalk.amazonaws.com"),
+                    ManagedPolicies = new[]
+                    {
+                        ManagedPolicy.FromAwsManagedPolicyName("AWSElasticBeanstalkManagedUpdatesCustomerRolePolicy")
+                    }
+                });
```
Comment (ganeshnj) [inline src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppElasticBeanstalkLinux/AppStack.cs:109]: New created role and the old role, do they have same level of permissions.

**Review by ganeshnj — CHANGES_REQUESTED** [review]

File: src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppElasticBeanstalkLinux/AppStack.cs (hunk 2 above)
Comment (philasmar) [reply src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppElasticBeanstalkLinux/AppStack.cs:109]: They are 2 different roles with different permissions

File: src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppElasticBeanstalkLinux/Configurations/ElasticBeanstalkManagedPlatformUpdatesConfiguration.cs (hunk 1 above)
Comment (philasmar) [reply src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppElasticBeanstalkLinux/Configurations/ElasticBeanstalkManagedPlatformUpdatesConfiguration.cs:9]: I don't see a reason to do so. We don't support this object type in the cli and we need to pass it to CDK as string as well.

**Review by normj — APPROVED** [review]

**Review by ganeshnj — APPROVED** [review]

**Pull request merged**